src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -204,28 +220,57 @@ private void StartNewLoggingThread()
                 Name = _name + " Thread"
             };
             _loggingThread.Start();
+            if (compress)
+            {
+                _compressLoggingThread = new Thread(BackgroundLoggerCompress)
+                {
+                    IsBackground = true,
+                    Name = _name + "Log Compression Thread"
+                };
+                _compressLoggingThread.Start();
+            }
+            else
+            {
+                _compressLoggingThread = null;
+            }
         }
 
         public void EndLogging()
         {
             _keepLogging.Lower();
             _hasEntries.Set();
+            _readyToCompress.Set();
+            _tokenSource.Cancel();
             _loggingThread.Join(TimeToWaitForLoggingToEndInMilliseconds);
-
+            _compressLoggingThread?.Join();
         }
 
         private FileStream GetNewStream(long maxFileSize)
         {
+            string[] logFiles;
+            string[] logGzFiles;
+            try
+            {
+                logFiles = Directory.GetFiles(_path, $"{_dateString}.*.log");
+                logGzFiles = Directory.GetFiles(_path, $"{_dateString}.*.log.gz");
+            }
+            catch (Exception)
+            {
+                // Something went wrong we will try again later
+                return null;
```

Comment:
This will cause NREs. We need to continue the execution of this method, but ignore the cleanup parts that use both of those variables